core/src/gl/shaderProgram.cpp

```diff
@@ -284,13 +284,10 @@ std::string ShaderProgram::applySourceBlocks(const std::string& source, bool fra
 
     auto sourcePos = source.begin();
 
-    auto linesBegin = std::sregex_iterator(source.begin(), source.end(), pragmaLine);
-    auto linesEnd = std::sregex_iterator();
+    for (auto it = source.begin(); std::regex_search(it, source.end(), sm, pragmaLine);) {
+        it += sm.position() + sm.length();
```

Comment:
Did we not just check that the `smatch::position` is different in earlier versions of gcc? Is it safe to relly on it then?

Or maybe it gives different results when used with regex_iterators!